### PR TITLE
SH-10: Handle backend errors in NGRX Effects

### PR DESCRIPTION
**Pull request by krucsodaniel**
## JIRA ticket link:
https://shoperz.atlassian.net/browse/SH-10

### CODE REVIEW

File: src/shared/enums/toast-message.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export enum ToastMessage {
+  addedToCart = 'cart.productAddedToCart',
+  amountUpdated = 'cart.productAmountUpdated',
+  alreadyInCart = 'cart.productIsAlreadyInCart',
+  removedFromCart = 'cart.productRemovedFromCart',
+}
```
Comment (bodnya29179) [inline src/shared/enums/toast-message.ts]: you shouldn't put the i18n messages to enum, it's extra

File: src/shared/services/toast-facade.service.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+@Injectable()
+export class ToastFacadeService {
+  private readonly toastConfigObject = {
+    positionClass: 'toast-bottom-center',
+  };
+
+  constructor(private toastr: ToastrService, private translate: TranslateService) {}
+
+  productAddedToCartMessage(): void {
```
Comment (bodnya29179) [inline src/shared/services/toast-facade.service.ts]: align the code in methods

Comment (bodnya29179) [reply src/shared/services/toast-facade.service.ts]: 1. Remove the "Facade" suffix in the service name.
2. Your toast methods should be reusable and shouldn't depend on the message. Make it reusable.

File: src/shared/store/cart/cart.effects.ts [hunk 3]
```diff
@@ -43,13 +46,14 @@ export class CartEffects {
       switchMap((action) => {
         return this.cartService.updateCart(action.id, action.amount)
           .pipe(
-            map((cartItem: ICartItem) =>
-              CartActions.productAmountUpdated({ cartItem })
-            ),
+            map((cartItem: ICartItem) => {
+              this.toastFacadeService.productAmountUpdatedMessage();
```
Comment (bodnya29179) [inline src/shared/store/cart/cart.effects.ts]: move to tap()

Comment (bodnya29179) [reply src/shared/store/cart/cart.effects.ts]: And what's the goal of using it there?

File: src/shared/store/cart/cart.effects.ts [hunk 4]
```diff
@@ -62,19 +66,21 @@ export class CartEffects {
 
         return this.cartService.removeProductFromCartById(id)
           .pipe(
-            map(() =>
-              CartActions.productRemovedFromCart({ id })
-            ),
+            map(() => {
+              this.toastFacadeService.productRemovedFromCartMessage();
```
Comment (bodnya29179) [inline src/shared/store/cart/cart.effects.ts]: and here too

File: src/shared/store/cart/cart.effects.ts (hunk 3 above)
Comment (krucsodaniel) [reply src/shared/store/cart/cart.effects.ts]: > And what's the goal of using it there?

Users can see that the amount of the product in the cart was updated

File: src/shared/services/toast.service.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { ToastrService } from 'ngx-toastr';
+import { TranslateService } from '@ngx-translate/core';
+
+@Injectable()
+export class ToastService {
+  private readonly toastConfigObject = {
+    positionClass: 'toast-bottom-center',
+  };
+
+  constructor(private toastr: ToastrService, private translate: TranslateService) {}
```
Comment (bodnya29179) [inline src/shared/services/toast.service.ts]: Let's make a simple service here.
1. Remove the translate service here.
2. Move it to places where you're passing the messages.
3. Passed messages should be already translated.
4. Create 4 methods in this service - show[Success/Info/Error/Warting]Toas(message: string)

File: src/shared/store/cart/cart.effects.ts [hunk 6]
```diff
@@ -26,13 +27,15 @@ export class CartEffects {
...
           .pipe(
-            map((cartItem: ICartItem) =>
-              CartActions.productAddedToCart({ cartItem })
-            ),
-            catchError((error) =>
-              of(CartActions.errorCart({ error }))
-            )
-          )
+            map((cartItem: ICartItem) => {
+              this.toastFacadeService.showToastMessage('productAddedToCart');
```
Comment (bodnya29179) [inline src/shared/store/cart/cart.effects.ts]: move to tap()

File: src/shared/store/cart/cart.effects.ts [hunk 7]
```diff
@@ -62,19 +62,17 @@ export class CartEffects {
...
+            catchError((error) => of(CartActions.errorCart({ error })))
+          );
       })
     )
   );
 
   constructor(
     private actions$: Actions,
     private cartService: CartService,
+    private toastFacadeService: ToastService,
```
Comment (bodnya29179) [inline src/shared/store/cart/cart.effects.ts]: rename toast facade to toast service

**Review by WeszeRita — COMMENTED** [review]
Nice job! :)

**Review by WeszeRita — APPROVED** [review]
Approved!

**Pull request merged**